### PR TITLE
fix: infinite loop in index writer

### PR DESCRIPTION
**Pull request by kolesnikovae**
Manual backporting of https://github.com/prometheus/prometheus/pull/12505. Thanks @pracucci!

I only wonder why it took so long to surface.

### CODE REVIEW

File: pkg/model/labels_test.go [hunk 1]
```diff
@@ -68,6 +68,42 @@ func TestLabelsUnique(t *testing.T) {
...
+		t.Run(test.name, func(t *testing.T) {
+			builder := NewLabelsBuilder(test.input)
+			for _, l := range test.add {
+				builder.Set(l.Name, l.Value)
+			}
+			assert.Equal(t, test.expected, builder.Labels())
+		})
+	}
+}
+
```
Comment (kolesnikovae) [inline pkg/model/labels_test.go:106]: I do not see any practical sense in fixing `LabelsBuilder` we borrowed from prometheus. Instead, in our case, a reusable map would work way better (and I'm 99.(9)% sure, more efficiently).

To be addressed in a separate PR.

image

File: pkg/phlaredb/labels/labels.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ func CreateProfileLabels(enforceOrder bool, p *profilev1.Profile, externalLabels
 		} else {
 			sort.Sort(lbs)
 		}
+		lbs = lbs.Unique()
```
Comment (kolesnikovae) [inline pkg/phlaredb/labels/labels.go:63]: Just in case: `Unique` is order-agnostic

**Comment by kolesnikovae** [comment]
I'm going to add an e2e test

**Review by korniltsev — APPROVED** [review]
